File: src/models/service_delivery_info.rs

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "PascalCase")]
pub struct ServiceDeliveryInfo {
    pub response_timestamp: Option<String>,
    pub producer_ref: String,
    pub request_message_identifier: Option<String>,
    pub response_message_ref: Option<String>
}

File: go/models/service_delivery_info.go
package models


type ServiceDeliveryInfo struct {
	ResponseTimestamp        time.Time `json:"ResponseTimestamp,omitempty"`
	ProducerRef              string    `json:"ProducerRef,omitempty"`
	RequestMessageIdentifier any       `json:"RequestMessageIdentifier,omitempty"`
	ResponseMessageRef       any       `json:"ResponseMessageRef,omitempty"`
} `json:"ServiceDeliveryInfo,omitempty"`